user/views.py

```diff
@@ -1,3 +1,60 @@
-from django.shortcuts import render
+import json
+import bcrypt
+import jwt
+
+from django.http  import JsonResponse
+from django.views import View
+
+from dienstag.my_settings import SECRET, ALGORITHM
+from .models              import User, UserInformation
+from .utils               import login_decorator
+
+class SignUpView(View):
+    def post(self, request):
+        MINIMUM_PASSWORD_LENGTH = 8
+        data = json.loads(request.body)
+        try:
+            email = data['email']
+            is_anonymous = data['is_anonymous']
+            if User.objects.filter(email=email).exists():
+                return JsonResponse({'message': 'ALREADY EXIST'}, status=400)
+
+            if '@' not in email or '.' not in email:
+                return JsonResponse({'message': 'INVALID_EMAIL'}, status=400)
+            
+            user = User.objects.create(email=email, is_anonymous=False)
+            
+            if not user.is_anonymous:
+                username = data['username']
+                password = data['password']
+                hashed_password = bcrypt.hashpw(password.encode('utf-8'), bcrypt.gensalt()).decode('utf-8')
+                phone_number = data['phone_number']
+
+                if len(password) < MINIMUM_PASSWORD_LENGTH:
+                    return JsonResponse({'message': 'PASSWORD_VALIDATION_ERROR'}, status=400)
+                    
+                UserInformation.objects.create(user=user, username=username, password=hashed_password, phone_number=phone_number)
+            return JsonResponse({'message': 'SUCCESS'}, status=200)
+
+        except KeyError:
+            return JsonResponse({'message':'KEY ERROR'}, status=400)
+
+class SignInView(View):
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+            email = data['email']
+            password = data['password']
+
+            if not User.objects.filter(email=email).exists():
+                return JsonResponse({'message': 'INVALID_USER'}, status=401)
+            user = User.objects.get(email=email)
+
+            if bcrypt.checkpw(password.encode('utf-8'), user.user_information.get(user=user).password.encode('utf-8')):
```

Comment:
#
```suggestion
            if bcrypt.checkpw(password.encode('utf-8'), user.user_information.password.encode('utf-8')):
```
user는 이미 위에서 email을 이용해 특정 user를 선택 했습니다. 따라서 get을 통해 다시 한 번 user를 선택할 필요는 없습니다.

Reply:
get을 지우고 values로 변경하였습니다~~